### PR TITLE
COMPAT: Adjust is_dataframe_like

### PR DESCRIPTION
**Pull request by TomAugspurger**
Pandas is adding a DataFrame.value_counts. Adjust accordingly.

Closes https://github.com/dask/dask/issues/5109

### CODE REVIEW

**Comment by quasiben** [comment]
Validated this works with cudf:


```python
In [1]: import cudf                                                                                                                                                                   

In [2]: df = cudf.DataFrame({'a': range(10), 'b': range(10)})                                                                                                                         
   
In [3]: typ = type(df)                                                                                                                                                                

In [4]: ( 
   ...: all(hasattr(typ, name) for name in ("groupby", "head", "merge", "mean")) 
   ...: and all(hasattr(df, name) for name in ("dtypes", "columns")) 
   ...: and not any(hasattr(typ, name) for name in ("name", "dtype")) 
   ...: )                                                                                                                                                                             
Out[4]: True   
```

File: dask/dataframe/tests/test_utils_dataframe.py [hunk 1]
```diff
@@ -376,9 +376,14 @@ def test_check_meta_typename():
...
 
 
-def test_is_dataframe_like():
+def test_is_dataframe_like(monkeypatch):
     df = pd.DataFrame({"x": [1, 2, 3]})
     ddf = dd.from_pandas(df, npartitions=1)
+
+    if not hasattr(pd.DataFrame, "value_counts"):
+        # TODO: Set the pandas version this shows up in (likely 1.0)
+        monkeypatch.setattr(pd.DataFrame, "value_counts", lambda x: None, raising=False)
```
Comment (jrbourbeau) [inline dask/dataframe/tests/test_utils_dataframe.py]: Do we want to keep this `monkeypatch` long term, or is it here only to demonstrate this test will pass once https://github.com/pandas-dev/pandas/pull/27350 is merged? The changes to `is_dataframe_like()` look to be backwards compatible

Comment (TomAugspurger) [reply dask/dataframe/tests/test_utils_dataframe.py]: Long term, we should remove it.

It occurs to me that we should test both with and without the patch. I can get to that next week unless someone beats me to it.

**Comment by jcrist** [comment]
Is this good to merge?

**Comment by TomAugspurger** [comment]
https://github.com/dask/dask/pull/5143#discussion_r306600188 Should be addressed. Will do that tomorrow.

**Comment by TomAugspurger** [comment]
Should be good to go now.

**Comment by jrbourbeau** [comment]
Thanks @TomAugspurger!

**Pull request merged**